doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+
+Example 2 : Replica Creation with ``RunSettings``, ``BatchSettings`` and `replicas`
+    To launch the Ensemble of identical Models as a batch job, begin by initializing a ``BatchSettings``
+    object:
+
+    .. code-block:: python
+
+        batch_args = {
+            "distribution": "block"
+            "exclusive": None
+        }
+        bs = exp.create_batch_settings(nodes=4,
+                               time="10:00:00",
+                               batch_args=batch_args)
+    The above ``BatchSettings`` object will tell SmartSim to run the Ensemble on four
+    nodes with a timeout of 10 hours.
+
+    Next, create a ``RunSettings`` object to expand to all Model replicas:
+
+    .. code-block:: python
+
+        rs = exp.create_run_settings(exe="python", exe_args="output_my_parameter.py")
+        rs.set_nodes(4)
+
+    Initialize the Ensemble by specifying the ``RunSettings`` object and number of clones to `replicas`:
+
+    .. code-block:: python
+
+        ensemble = exp.create_ensemble("ensemble-replica",
+                               replicas=4,
+                               run_settings=rs)
+
+    By passing in `replicas=4`, four identical Ensemble members will be initialized.
+
+
+---------------
+Manually Append
+---------------
+.. _append_init:
+Manually appending models involves the addition of user created model instances to an ensemble,
+offering a level of customization in ensemble design. This approach is favorable when users
+have distinct requirements for individual models, such as variations in parameters, run settings,
+or model architectures.
+
+Examples
+--------
+Example 1 : Append Models with ``BatchSettings``
+    To create an empty Ensemble to append Models, initialize the Ensemble with
+    a batch settings object:
+
+    .. code-block:: python
+
+        bs = exp.create_batch_settings(nodes=10,
+                               time="01:00:00")
+        ensemble = exp.create_ensemble("ensemble-append", batch_settings=bs)
+
+    Next, create the Models to append to the Ensemble:
+
+    .. code-block:: python
+
+        srun_settings_1 = exp.create_run_settings(exe=exe, exe_args="path/to/script_1")
+        srun_settings_2 = exp.create_run_settings(exe=exe, exe_args="path/to/script_2")
+        model_1 = exp.create_model(name="model_1", run_settings=srun_settings_1)
+        model_2 = exp.create_model(name="model_2", run_settings=srun_settings_2)
+
+    Finally, append the ``Model`` object to the ``Ensemble``:
+
+    .. code-block:: python
+
+        ensemble.add_model(model_1)
+        ensemble.add_model(model_2)
+=====
+Files
+=====
+.. _attach_files:
+--------
+Overview
+--------
+SmartSim enables users to attach files to Ensembles for use within the workflow
+through the ``Ensemble.attach_generator_file()`` function. The function
+accepts three input arguments:
+
+1. `to_copy` (list, optional): files to copy, defaults to [].
```

Comment:
- slightly tighter type hint is `list[str]`

- probably inconsequential, but the defaults are actually `None`